lib/Command/ListLibraries.php

```diff
@@ -0,0 +1,178 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\SharePoint\Command;
+
+use OCA\SharePoint\ClientFactory;
+use OCA\SharePoint\ContextsFactory;
+use Office365\PHP\Client\SharePoint\SPList;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+class ListLibraries extends Command {
+
+	/** @var ClientFactory */
+	protected $clientFactory;
+	/** @var ContextsFactory */
+	protected $ctxFactory;
+
+	public function __construct(ClientFactory $clientFactory, ContextsFactory $ctxFactory) {
+		parent::__construct();
+		$this->clientFactory = $clientFactory;
+		$this->ctxFactory = $ctxFactory;
+	}
+
+	protected function configure() {
+		$this
+			->setName('sharepoint:list-libraries')
+			->setDescription('List the available document libraries')
+			->addArgument(
+				'host',
+				InputArgument::REQUIRED,
+				'the url to the sharepoint server'
+			)
+			->addArgument(
+				'login',
+				InputArgument::REQUIRED,
+				'the login name to authenticate with'
+			)
+			->addArgument(
+				'password',
+				InputArgument::OPTIONAL,
+				'will be asked interactively if not provided as argument',
+				null
+			)
+			->addOption(
+				'all-properties',
+				null,
+				InputOption::VALUE_NONE,
+				'print all properties of each document library'
+			)
+			->addOption(
+				'json',
+				null,
+				InputOption::VALUE_NONE,
+				'output in JSON instead of a table'
+			);
+	}
+
+	protected function defaultOutput(OutputInterface $output, array $libraries, bool $json = false) {
+		$rows = [];
+		foreach ($libraries as $library) {
+			$mdate = new \DateTime($library->getProperties()['LastItemModifiedDate']);
+			$rows[] = [
+				'title' => $library->getProperties()['Title'],
+				'items' => $library->getProperties()['ItemCount'],
+				'mdate' => date('Y-m-d H:i', $mdate->getTimestamp()),
+			];
+		}
+
+		if($json) {
+			$output->writeln(\json_encode($rows));
+			return;
+		}
+
+		$table = new Table($output);
+		$table->setHeaders(['Title', 'Items', 'Last modification']);
+		$table->setRows($rows);
+		$table->render();
+
+	}
+
+	protected function allPropertiesOutput(OutputInterface $output, array $libraries, bool $json = false) {
+		if(empty($libraries)) {
+			return;
+		}
+
+		$rows = [];
+		$i = 0;
+
+		/** @var SPList $library */
+		foreach ($libraries as $library) {
+			$props = $library->getProperties();
+			$rows[$i] = [];
+
+			$rows[$i]['Title'] = $props['Title'];
```

Comment:
Why is this needed here? Wouldn't the foreach below do this anyways?

Reply:
to sort it on top

Reply:
Ah - got it 👍